src/main/java/com/ivanmostovyi/demo/service/MessageServiceImpl.java

```diff
@@ -1,47 +1,89 @@
 
-    private MessageRepository messageRepository;
+    private InboxMessageRepository inboxMessageRepository;
 
-    private MessageToMessageDtoMapper messageToMessageDtoMapper;
+    private OutboxMessageRepository outboxMessageRepository;
 
-    public MessageServiceImpl(UserRepository userRepository, MessageRepository messageRepository,
-                              MessageToMessageDtoMapper messageToMessageDtoMapper) {
+    private InboxMessageToInboxMessageDtoMapper inboxMessageToInboxMessageDtoMapper;
+
+    private OutboxMessageToOutboxMessageDtoMapper outboxMessageToOutboxMessageDtoMapper;
+
+    public MessageServiceImpl(UserRepository userRepository, InboxMessageRepository inboxMessageRepository,
+                              OutboxMessageRepository outboxMessageRepository,
+                              InboxMessageToInboxMessageDtoMapper inboxMessageToInboxMessageDtoMapper,
+                              OutboxMessageToOutboxMessageDtoMapper outboxMessageToOutboxMessageDtoMapper) {
 
         this.userRepository = userRepository;
-        this.messageRepository = messageRepository;
-        this.messageToMessageDtoMapper = messageToMessageDtoMapper;
+        this.inboxMessageRepository = inboxMessageRepository;
+        this.outboxMessageRepository = outboxMessageRepository;
+        this.inboxMessageToInboxMessageDtoMapper = inboxMessageToInboxMessageDtoMapper;
+        this.outboxMessageToOutboxMessageDtoMapper = outboxMessageToOutboxMessageDtoMapper;
     }
 
     @Async
     @Override
-    public void create(MessageFormDto messageFormDto, User senderUser){
+    public void create(MessageFormDto messageFormDto, User senderUser) {
+
+        String[] receiverUsernames = messageFormDto.getReceiverUsername().split(" ");
+
+        List<User> receiverUsers = new ArrayList<>();
+        List<String> notFoundUsernames = new ArrayList<>();
+
+        Arrays.stream(receiverUsernames).forEach(
+                username -> userRepository.findByUsername(username)
+                        .ifPresentOrElse(receiverUsers::add, () -> notFoundUsernames.add(username))
+        );
+
+        receiverUsers.forEach(receiverUser -> createInboxMessage(messageFormDto, receiverUser, senderUser));
+
+        OutboxMessage outboxMessage = OutboxMessage.builder()
+                .marked(false)
+                .receiverUsernames(receiverUsernames)
+                .title(messageFormDto.getTitle())
+                .body(messageFormDto.getBody())
+                .senderUserId(senderUser.getId())
+                .date(LocalDateTime.now())
+                .build();
 
-        User receiverUser = userRepository.findByUsername(messageFormDto.getReceiverUsername())
-                                          .orElseThrow(() -> new MessageSendingException("Receiver was not found!"));
+        outboxMessageRepository.save(outboxMessage);
 
-        Message message = Message.builder()
+        if (!notFoundUsernames.isEmpty()) {
```

Comment:
Throwing exception in successful flow is not very good.
For the design you choose, it's better to create Inbox message with sending error here.